src/pipeline/filter_funcs.c

```diff
@@ -0,0 +1,526 @@
+			return -DER_DIV_BY_ZERO;                                                   \
+		*res = left / right;                                                               \
+		return 0;                                                                          \
+	}
+
+arithfunc_div(u, uint64_t)
+arithfunc_div(i, int64_t)
+arithfunc_div(d, double)
+
+#define filter_func_arith(op, type, ctype)                                                         \
+	int filter_func_##op##_##type(struct filter_part_run_t *args)                              \
+	{                                                                                          \
+		_##ctype left  = (_##ctype)0;                                                      \
+		_##ctype right = (_##ctype)0;                                                      \
+		int      rc    = 0;                                                                \
+		rc = filter_func_getdata_##type(args, &left);                                      \
+		if (unlikely(rc != 0))                                                             \
+			D_GOTO(exit, rc);                                                          \
+		rc = filter_func_getdata_##type(args, &right);                                     \
+		if (unlikely(rc != 0))                                                             \
+			D_GOTO(exit, rc);                                                          \
+		rc = arithfunc_##op##_##type(left, right, &args->value_##type##_out);              \
+exit:                                                                                              \
+		return rc;                                                                         \
+	}
+
+filter_func_arith(add, u, uint64_t)
+filter_func_arith(add, i, int64_t)
+filter_func_arith(add, d, double)
+filter_func_arith(sub, u, uint64_t)
+filter_func_arith(sub, i, int64_t)
+filter_func_arith(sub, d, double)
+filter_func_arith(mul, u, uint64_t)
+filter_func_arith(mul, i, int64_t)
+filter_func_arith(mul, d, double)
+filter_func_arith(div, u, uint64_t)
+filter_func_arith(div, i, int64_t)
+filter_func_arith(div, d, double)
+
+#define filter_func_bitand(type, ctype)                                                            \
+	int filter_func_bitand_##type(struct filter_part_run_t *args)                              \
+	{                                                                                          \
+		_##ctype left  = 0;                                                                \
+		_##ctype right = 0;                                                                \
+		int      rc    = 0;                                                                \
+		rc = filter_func_getdata_##type(args, &left);                                      \
+		if (unlikely(rc != 0))                                                             \
+			D_GOTO(exit, rc);                                                          \
```

Comment:
no need to use goto here and can just return rc?

Reply:
right